ui/send_page.go

```diff
@@ -1,21 +1,724 @@
+					}
+					inset.Layout(gtx, func() {
+						pg.theme.Body1(fmt.Sprintf("Spendable: %s", dcrutil.Amount(account.SpendableBalance).String())).Layout(gtx)
+					})
+				})
+				pointer.Rect(image.Rectangle{Max: gtx.Dimensions.Size}).Add(gtx.Ops)
+				pg.accountSelectorButtons[buttonKey].Layout(gtx)
+			})
 		}
-		win.combined.sel.Options = strs
+	}
+
+	list := layout.List{Axis: layout.Vertical}
+	pg.theme.Modal(gtx, func() {
+		list.Layout(gtx, len(w), func(i int) {
+			layout.UniformInset(unit.Dp(0)).Layout(gtx, w[i])
+		})
+	})
+}
 
-		win.combined.sel.Layout(win.gtx, func() {
+func (pg *Send) drawConfirmationModal(gtx *layout.Context) {
+	widgetsWrapper := [][]func(){
+		{
+			func() {
+				pg.theme.H4("Confirm to send").Layout(gtx)
+			},
+			func() {
+				if pg.sendErrorLabel.Text != "" {
+					pg.sendErrorLabel.Layout(gtx)
+				}
+			},
+			func() {
+				gtx.Constraints.Width.Min = gtx.Constraints.Width.Max
+				layout.Stack{Alignment: layout.Center}.Layout(gtx,
+					layout.Expanded(func() {
+						pg.theme.Body1(fmt.Sprintf("Sending from %s (%s)", pg.selectedAccountLabel.Text, pg.selectedWalletLabel.Text)).Layout(gtx)
+					}),
+				)
+			},
+			func() {},
+			func() {
+				gtx.Constraints.Width.Min = gtx.Constraints.Width.Max
+				layout.Stack{Alignment: layout.Center}.Layout(gtx,
+					layout.Expanded(func() {
+						pg.confirmDestinationAddressLabel.Layout(gtx)
+					}),
+				)
+			},
+			func() {
+				gtx.Constraints.Width.Min = gtx.Constraints.Width.Max
+				layout.Stack{Alignment: layout.Center}.Layout(gtx,
+					layout.Expanded(func() {
+						pg.theme.Body1(pg.destinationAddressEditor.editor.Text()).Layout(gtx)
+					}),
+				)
+			},
+		},
+		pg.getTransactionDetailWidgets(gtx),
+		{
+			func() {
+				gtx.Constraints.Width.Min = gtx.Constraints.Width.Max
+				layout.Stack{Alignment: layout.Center}.Layout(gtx,
+					layout.Expanded(func() {
+						pg.confirmWarningLabel.Layout(gtx)
+					}),
+				)
+			},
+			func() {
+				layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
+					layout.Rigid(func() {
+						for pg.confirmButton.button.Clicked(gtx) {
+							pg.isPasswordModalOpen = true
+						}
+						pg.confirmButton.layout(gtx)
+					}),
+					layout.Rigid(func() {
+						for pg.closeConfirmationModalButton.button.Clicked(gtx) {
+							pg.isConfirmationModalOpen = false
+						}
+						pg.closeConfirmationModalButton.layout(gtx)
+					}),
+				)
+			},
+		},
+	}
 
+	widgets := reduceWidgetsWrapper(widgetsWrapper)
+	pg.theme.Modal(gtx, func() {
+		(&layout.List{Axis: layout.Vertical}).Layout(gtx, len(widgets), func(i int) {
+			layout.UniformInset(unit.Dp(10)).Layout(gtx, widgets[i])
 		})
+	})
+}
+
+func (pg *Send) setWallets(wallets []wallet.InfoShort) {
+	pg.wallets = wallets
+
+	for i := range wallets {
+		if len(wallets[i].Accounts) == 0 {
+			continue
+		}
+
+		pg.setSelectedAccount(wallets[i], wallets[i].Accounts[0])
+		pg.balanceAfterSendValueLabel.Text = dcrutil.Amount(pg.selectedWallet.SpendableBalance).String()
```

Comment:
"SpendableBalance" should be preprocessed from the wallet method. You shouldn't need to process values in the UI.